File: backend_api1/models/order.js
const mongoose = require("mongoose");

const orderSchema = mongoose.Schema({
  fullName: {
    type: String,
    require: true,
  },

  email: {
    type: String,
    require: true,
  },

  state: {
    type: String,
    require: true,
  },

  city: {
    type: String,
    require: true,
  },

  locality: {
    type: String,
    require: true,
  },

  productName: {
    type: String,
    require: true,
  },

  productPrice: {
    type: Number,
    require: true,
  },

  quantity: {
    type: Number,
    require: true,
  },

  category: {
    type: String,
    require: true,
  },

  image: {
    type: String,
    require: true,
  },

  buyerId: {
    type: String,
    require: true,
  },

  vendorId: {
    type: String,
    require: true,
  },

  processing: {
    type: Boolean,
    default: true,
  },

  delivered: {
    type: Boolean,
    default: false,
  },

  createdAt: {
    type: Number,
    require: true,
  },
});
const Order = mongoose.model("Order", orderSchema);
module.exports = Order;

File: demo2/lib/models/order.dart
import 'dart:convert';

class Order {
  final String id;
  final String fullName;
  final String email;
  final String state;
  final String city;
  final String locality;
  final String productName;
  final int productPrice;
  final int quantity;
  final String category;
  final String image;
  final String buyerId;
  final String vendorId;
  final bool processing;
  final bool delivered;

  Order({
    required this.id,
    required this.fullName,
    required this.email,
    required this.state,
    required this.city,
    required this.locality,
    required this.productName,
    required this.productPrice,
    required this.quantity,
    required this.category,
    required this.image,
    required this.buyerId,
    required this.vendorId,
    required this.processing,
    required this.delivered,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'fullName': fullName,
      'email': email,
      'state': state,
      'city': city,
      'locality': locality,
      'productName': productName,
      'productPrice': productPrice,
      'quantity': quantity,
      'category': category,
      'image': image,
      'buyerId': buyerId,
      'vendorId': vendorId,
      'processing': processing,
      'delivered': delivered,
    };
  }

  String toJson() => json.encode(toMap());

  factory Order.fromJson(Map<String, dynamic> map) {
    return Order(
      id: map['_id'] as String,
      fullName: map['fullName'] as String,
      email: map['email'] as String,
      state: map['state'] as String,
      city: map['city'] as String,
      locality: map['locality'] as String,
      productName: map['productName'] as String,
      productPrice: map['productPrice'] as int,
      quantity: map['quantity'] as int,
      category: map['category'] as String,
      image: map['image'] as String,
      buyerId: map['buyerId'] as String,
      vendorId: map['vendorId'] as String,
      processing: map['processing'] as bool,
      delivered: map['delivered'] as bool,
    );
  }
}
